whole-tale/ngx-dashboard
# Public WholeTale Dashboard access as anonymous user

## Problem
It would be nice to offer a read-only version of the Tale Catalog / Run Tale views. This would allow new users to preview the system before trying it out.

## Approach
* Adjust AuthGuard to allow unauthed users in Catalog/Run views
* Add a new button to the Login view to **Preview WholeTale** (note: button location/styling is subject to change)
* Disable some of the features that unauthed users can't use
    * Hide Shared/My Tales tabs on Catalog
    * Hide "Duplicate Tale" dropdown item
    * Hide Notification Stream
    * Hide link to Settings page
    * Hide Home Folder
    * Disable Run Tale button (w/ tooltip)

<img width="1680" alt="Screen Shot 2022-06-24 at 5 21 39 PM" src="https://user-images.githubusercontent.com/1413653/175701263-8d02a5d1-362e-4459-814b-65ca67530f3d.png">

<img width="1660" alt="Screen Shot 2022-06-06 at 2 35 26 PM" src="https://user-images.githubusercontent.com/1413653/172234307-6d0b1930-35cf-4c40-95db-137c296b20f6.png">

<img width="1665" alt="Screen Shot 2022-06-06 at 2 36 29 PM" src="https://user-images.githubusercontent.com/1413653/172234517-13f02159-d4b5-4176-80e4-1881090c40a2.png">

## How to Test
Prerequisites: at least one Tale created and marked as "Public" 

1. Checkout this branch locally, rebuild the dashboard
2. Close all incognito windows, then open a new incognito window
3. Navigate to the WholeTale Dashboard
    * You should see a Read-Only version of the Public Tales Catalog, even without needing to log in
    * You should see that the "Run Tale button has been hidden on each Tale
    * You should see a "Sign In with Globus" button at the top-right
4. Click the title of the Tale or hover over the image and click the View button
    * You should be brought to a Read-Only version of the Run Tale view
    * You should see that the Run Tale button has been disabled (but offers a tooltip informing the user that they need to log in)
    * You should see that the "Interact" and "Share" tabs are not shown
6. Click the Files tab
    * You should see a Read-Only version of the file-explorer
    * You should see that the "Home" folder nav is not shown
7. Expand the dropdown menu at the top-right in the tale banner
    * You should see only "Read the docs" listed here


## TODOs
* Consistency between "Run Tale" button on Catalog view and on Run view
* ~~GET version and recorded runs are both returning a 401 when not logged in.. does this make sense?~~
